networkx/algorithms/tests/test_euler.py

```diff
@@ -80,18 +77,79 @@ def test_eulerian_circuit_digraph(self):
         assert_equal(nodes,[1,2,3,0])
         assert_equal(edges,[(1,2),(2,3),(3,0),(0,1)])
 
-
     def test_eulerian_circuit_multigraph(self):
         G=nx.MultiGraph()
         G.add_cycle([0,1,2,3])
         G.add_edge(1,2)
         G.add_edge(1,2)
         edges=list(eulerian_circuit(G,source=0))
-        nodes=[u for u,v in edges]
+        nodes=[u for u, v in edges]
         assert_equal(nodes,[0,3,2,1,2,1])
         assert_equal(edges,[(0,3),(3,2),(2,1),(1,2),(2,1),(1,0)])
 
+    @raises(nx.NetworkXError)
+    def test_not_eulerian(self):
+        next(list(eulerian_circuit(nx.complete_graph(4))))
+
+
+class TestEulerianPath:
+
+    # Use the bridges of Königsberg
+    bridges = nx.MultiGraph([('W', 'N'), ('W', 'N'), ('N', 'E'), ('E', 'W'),
```

Comment:
This sort of thing usually appears in the `setup(self)` instance method of the test class.